tests/cupy_tests/core_tests/test_raw.py

```diff
@@ -973,20 +973,25 @@ def test_compile_module(self):
 #include <cooperative_groups.h>
 
 extern "C" __global__
-void test_grid_sync(const float* x1, const float* x2, float* y) {
+void test_grid_sync(const float* x1, const float* x2, float* y, int n) {
     namespace cg = cooperative_groups;
     cg::grid_group grid = cg::this_grid();
     int size = gridDim.x * blockDim.x;
     int tid = blockDim.x * blockIdx.x + threadIdx.x;
-    y[tid] = x1[tid];
+    for (int i = tid; i < n; i += size) {
+        y[i] = x1[i];
+    }
     cg::sync(grid);
-    y[size - tid - 1] += x2[size - tid - 1];
+    for (int i = n - 1 - tid; i >= 0; i -= size) {
```

Comment:
Just curious, what's the rationale to run the loop backwards?

Reply:
This is to make sure that grid synchronization is working properly.

This sample code first assigns the elements of array `x1` to array `y` and then adds the elements of aray x2 to it. If you look at it on a per-element basis, assignment and addition are done by different threads, so the result can be weired if grid synchronization is not working properly (though this was not written by me..).